### DISCUSSION TITLE
Deep relation filter error

### DISCUSSION THREAD

**Discussion by fatoon29** [directus/directus#8491] [Archive]
Hello.
I have a problem when i try to filter some table with m2m filter.

The project : 
It's an application to manage radio. 
I have multiple radio (radio) in the app. There are programs (radio_program) link to radio by a m2m field (radio_program_radio). After there are episodes (radio_episode), link to some program by a m2o field (program in the episode table).
Each member is assigned to radio by a m2m field to have permission to manage radio, program and episodes in the radio he is assigned to.
You can find the extract of the table in the image.

The problem is when i want to call radio_episode filter by user permission, i have an internal server error.

My request is : 

`localhost:8055/items/radio_episode?filter={"program":{"radio":{"radio_id":{"permission":{"permission_id":{"user":{"_eq":"d4265027-0519-47c1-a31e-e6073a7aee0c"}}}}}}}`

The response of the request is :

```
{
  "errors": [
    {
      "message": "select `radio_episode`.`id`, `radio_episode`.`status`, `radio_episode`.`user_created`, `radio_episode`.`date_created`, `radio_episode`.`user_updated`, `radio_episode`.`date_updated`, `radio_episode`.`date`, `radio_episode`.`image`, `radio_episode`.`program`, `radio_episode`.`podcast`, `radio_episode`.`duration`, `radio_episode`.`title_breton`, `radio_episode`.`description_breton`, `radio_episode`.`title_gallo`, `radio_episode`.`description_gallo`, `radio_episode`.`translation_french`, `radio_episode`.`translation_breton`, `radio_episode`.`translation_gallo`, `radio_episode`.`title_french`, `radio_episode`.`description_french`, `radio_episode`.`date_scheduled`, `radio_episode`.`sync_wp_token` from `radio_episode` left join `radio_program` as `ujggt` on `radio_episode`.`program` = `ujggt`.`id` left join `radio_program_radio` as `ohnij` on `ujggt`.`id` = `ohnij`.`radio_program_id` where `permission`.`user` = 'd4265027-0519-47c1-a31e-e6073a7aee0c' order by `radio_episode`.`id` asc limit 100 - ER_BAD_FIELD_ERROR: Unknown column 'permission.user' in 'where clause'",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"
      }
    }
  ]
}
```
It Looks like there is a left join missing : radio_permission 

But when a try to call radio_program filter by user permission, the request success

`20.0.0.227:8055/items/radio_program?filter={"radio":{"radio_id":{"permission":{"permission_id":{"user":{"_eq":"d4265027-0519-47c1-a31e-e6073a7aee0c"}}}}}}`

![FireShot Capture 025 - localhost_8056 _ database _ radiobreizh - phpMyAdmin 5 1 1 - localhost](https://user-images.githubusercontent.com/55839100/135626947-488a951e-98a6-4873-ac51-490e42f1189a.png)
e.

**Comment by rijkvanzanten** [comment]
I believe this is a duplicate of https://github.com/directus/directus/issues/5747

PS Issues is a better place for bugs in the platform.  This is more for feature requests/ Q&A  🙂

**Reply by fatoon29** [reply]
Hi, 
Next times i'll go to issues ^^
I'm following the #5747 issues, thanks
